src/common.rs

```diff
@@ -4,6 +4,91 @@
 
 use arrayvec::ArrayVec;
 
+/// Defines a trait that chooses between libstd or libm implementations of float methods.
+macro_rules! define_float_funcs {
+    ($(
+        fn $name:ident(self $(,$arg:ident: $arg_ty:ty)*) -> $ret:ty
+        => $lname:ident/$lfname:ident;
+    )+) => {
+        pub(crate) trait FloatFuncs : Sized {
+            $(fn $name(self $(,$arg: $arg_ty)*) -> $ret;)+
+        }
+
+        impl FloatFuncs for f32 {
+            $(fn $name(self $(,$arg: $arg_ty)*) -> $ret {
+                #[cfg(feature = "std")]
```

Comment:
slightly pedantic, but if we *do* want to use this trait in both cases (which would require us to call the trait methods directly, or give them names that do not shadow the std names) I would do this the other way around: if `libm` is set, always use libm, if `libm` is not set and `std` is set use `std`. (Because perhaps for some reason you want to use musl even if std is available? Tell me if I'm wrong 💁)